src/vmm/src/vstate/system.rs

```diff
@@ -50,7 +65,9 @@ pub struct KvmContext {
 impl KvmContext {
     pub fn new() -> Result<Self> {
         use kvm_ioctls::Cap::*;
-        let kvm = Kvm::new().expect("Error creating the Kvm object");
+        let kvm = Kvm::new().unwrap_or_else(|e| {
```

Comment:
as discussed, let's do a: `Kvm::new().map_err(Error::KvmInit)?;`

Please also confirm by running manually that the right error message is outputted

Reply:
I have indeed run it and update the overview for the PR as well. I will need to re-test once as soon as I make the discussed changes.